### PR TITLE
Favorite Movie Order Fix

### PR DESCRIPTION
**Pull request by mncadavid**

### CODE REVIEW

File: controllers/movies.js [hunk 1]
```diff
@@ -217,6 +217,7 @@ const renderFavoritesPageFiltered  = (req, res) => {
             let favoriteMovies = [];
             let favoriteCountArray = [];
             let favoritePromises = [];
+            let addedMovies = 0;
```
Comment (Steamjunk) [inline controllers/movies.js:220]: Why did you need this variable here and not in the above method?

Comment (mncadavid) [reply controllers/movies.js:220]: I only need this variable when the results are actually filtered because when we are adding movies to the array to be passed to the ejs we are skipping over some of the movies.  In the other method, I knew that the array length would be equal to the number of movies returned but now I don't know the array length after the movies are filtered by genre.
